### PR TITLE
chore: Adds identifier assertions for xhr and fetch adapters

### PR DESCRIPTION
**Pull request by jasonmit**
Partially solves #244 

Puppeteer I'll tackle in a follow up PR.

### CODE REVIEW

File: tests/integration/adapter-identifier-tests.js [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+      });
+
+      testConfiguration('url.username', false, {
+        fetch: {
+          id: '3dbff3c3fbccd6f97d01be31fc7fdd59',
+          identifiers: {
+            headers: { 'content-type': 'application/json;charset=utf-8' },
+            method: 'POST',
+            body: '{}',
+            url: 'http://localhost:4000/pathname?query=param'
```
Comment (jasonmit) [inline tests/integration/adapter-identifier-tests.js]: Looks like we have a bug in Polly, url here dropped `password` from the matchBy.  Here we only turned off username, but password went with it.

File: tests/integration/adapter-identifier-tests.js [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+          }
+        },
+        xhr: {
+          id: '3dbff3c3fbccd6f97d01be31fc7fdd59',
+          identifiers: {
+            headers: { 'content-type': 'application/json;charset=utf-8' },
+            method: 'POST',
+            body: '{}',
+            url: 'http://localhost:4000/pathname?query=param'
```
Comment (jasonmit) [inline tests/integration/adapter-identifier-tests.js]: Same as above

File: tests/integration/adapter-identifier-tests.js [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+          body: undefined,
+          headers: { 'content-type': 'application/json;charset=utf-8' },
+          method: 'POST',
+          url: 'http://a:b@localhost:4000/pathname?query=param'
+        }
+      },
+      xhr: {
+        id: '3dbfc69c4a73f3f368ae00aee80d2379',
+        identifiers: {
+          body: undefined,
```
Comment (jasonmit) [inline tests/integration/adapter-identifier-tests.js]: I don't think `body` should be set on the Map but not a big deal

File: tests/integration/adapter-identifier-tests.js [hunk 4]
```diff
@@ -0,0 +1,308 @@
+import set from 'lodash-es/set';
+
+function adapterIdentifierTests() {
```
Comment (jasonmit) [inline tests/integration/adapter-identifier-tests.js]: The general id here is if we change any of the identifiers, we either introduced a breaking change or a bug.  In writing these tests, I did spot existing bug - there could be more.

File: tests/integration/adapter-identifier-tests.js [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+  describe.only('matchRequestsBy', function() {
```
Comment (jasonmit) [inline tests/integration/adapter-identifier-tests.js]: Remove once I'm done with all of this

File: tests/integration/adapter-identifier-tests.js [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+          identifiers: {
+            headers: { 'content-type': 'application/json;charset=utf-8' },
+            method: 'POST',
+            body: '{}',
+            url: 'http://a:b@localhost/pathname?query=param'
+          }
+        },
+        xhr: {
+          id: '147ca2615501f226981619c32341699b',
+          identifiers: {
```
Comment (jasonmit) [inline tests/integration/adapter-identifier-tests.js]: Decided to assert also against the identifiers, will help when we need to triage why an ID changed.

File: tests/integration/adapter-identifier-tests.js (hunk 1 above)
Comment (offirgolan) [reply tests/integration/adapter-identifier-tests.js]: Good catch. We shoud create a new issue with this and follow up on this unless you have the time to fix it with this PR.

**Comment by offirgolan** [comment]
@jasonmit this looks great! Thanks for taking this up 😄.

**Comment by jasonmit** [comment]
Awesome, I'll wrap this up over the weekend!

File: tests/integration/adapter-identifier-tests.js (hunk 1 above)
Comment (jasonmit) [reply tests/integration/adapter-identifier-tests.js]: Looks like the spec doesn't support passing credentials in the URL - results in a TypeError.  I think we should deprecate and eventually remove url.password and url.username options.

**Pull request merged**